### ISSUE TITLE
Private Endpoint - Unnecessary variables?

### ISSUE THREAD

**Issue by MariusStorhaug** [Azure/ResourceModules#1255]
## Description

There are two variables in private endpoint that is not really required by the looks of it. Its a variable not really used as a whole, while its properties are. Why are we not just linking those directly? is it to have readable names where we use them (Clean code)?

Reference to an example of this
https://github.com/Azure/ResourceModules/blob/317582ac48847fcadbfb925d5888939f240118f2/arm/Microsoft.KeyVault/vaults/.bicep/nested_privateEndpoint.bicep#L7

**Comment by MrMCake** [comment]
(Untested) suggestion in `nested_privateEndpoint.bicep`:

```bicep
@description('Required. The resource ID of the resource to link the private endpoint to')
param privateEndpointResourceId string

@description('Optional. Tags to apply to the private endpoint')
param tags object = {}

@description('Optional. the location to deploy the private endpoint into')
param location string = resourceGroup().name

@description('Optional. The name of the private endpoint.')
param name string = '${last(split(privateEndpointResourceId, '/'))}-${groupIds[0]}'

@description('Required. The resource ID of a subnet to link the private endpoint to.')
param subnetResourceId string

@description('Optional. The name of the private DNS zone to create together with the endpoint')
param privateDNSZoneName string = '${name}/default'

@description('Optional. The ID(s) of the group(s) obtained from the remote resource that this private endpoint should connect to.')
param groupIds array

@description('Optional. A collection of private dns zone configurations of the private dns zone group.')
param privateDnsZoneResourceIds array = []

@description('Optional. An array of custom DNS configurations.')
param customDnsConfigs array = []

@description('Optional. A grouping of information about the connection to the remote resource. Used when the network admin does not have access to approve connections to the remote resource.')
param manualPrivateLinkServiceConnections array = []

resource privateEndpoint 'Microsoft.Network/privateEndpoints@2021-05-01' = {
  name: name
  location: location
  tags: tags
  properties: {
    privateLinkServiceConnections: [
      {
        name: name
        properties: {
          privateLinkServiceId: privateEndpointResourceId
          groupIds: groupIds
        }
      }
    ]
    manualPrivateLinkServiceConnections: manualPrivateLinkServiceConnections
    subnet: {
      id: subnetResourceId
    }
    customDnsConfigs: customDnsConfigs
  }
}

resource privateDnsZoneGroups 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2021-02-01' = if (!empty(privateDnsZoneResourceIds)) {
  name: privateDNSZoneName
  properties: {
    privateDnsZoneConfigs: [for resourceId in privateDnsZoneResourceIds: {
      name: last(split(resourceId, '/'))
      properties: {
        privateDnsZoneId: resourceId
      }
    }]
  }
}
```

(Untested) suggestion for parent (e.g. KeyVault) template:

```bicep
module keyVault_privateEndpoints '.bicep/nested_privateEndpoint.bicep' = [for (privateEndpoint, index) in privateEndpoints: {
  name: '${uniqueString(deployment().name, location)}-KeyVault-PrivateEndpoint-${index}'
  params: {
    privateEndpointResourceId: keyVault.id
    tags: tags
    location: empty(privateEndpoints) ? 'dummy' : reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
    groupIds: contains(privateEndpoint, 'groupIds') ? privateEndpoint.groupIds : [
      'vault'
    ]
    subnetResourceId: privateEndpoint.subnetResourceId
    customDnsConfigs: contains(privateEndpoint, 'customDnsConfigs') ? privateEndpoint.customDnsConfigs : []
    manualPrivateLinkServiceConnections: contains(privateEndpoint, 'manualPrivateLinkServiceConnections') ? privateEndpoint.manualPrivateLinkServiceConnections : []
    name: contains(privateEndpoint, 'name') ? privateEndpoint.name : '${last(split(keyVault.id, '/'))}-${contains(privateEndpoint, 'groupIds') ? privateEndpoint.groupIds[0] : 'vault'}'
    privateDNSZoneName: contains(privateEndpoint, 'privateDNSZoneName') ? privateEndpoint.privateDNSZoneName : '${name}/default'
    privateDnsZoneResourceIds: contains(privateEndpoint, 'privateDnsZoneResourceIds') ? privateEndpoint.privateDnsZoneResourceIds : []
  }
}]
```

(Untested) parameter file parameter:

```json
        "privateEndpoints": {
            "value": [
                {
                    "subnetResourceId": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/virtualNetworks/adp-<<namePrefix>>-az-vnet-x-001/subnets/<<namePrefix>>-az-subnet-x-005-privateEndpoints"
                }
            ]
        },
```

**Comment by eriqua** [comment]
I'd suggest to close this issue, indirectly solved by #905

**Comment by MrMCake** [comment]
Agreed
